# ML-02 — Research Question and Provisional Lane

[![Open In Colab](https://colab.research.google.com/assets/colab-badge.svg)](https://colab.research.google.com/github/arsenalmorimori/flyrank_ml_pipeline/blob/main/work/notebooks/w01_research_question.ipynb?flush_cache=true)

This skeleton is yours to fill. Work the sections **in order** — each one has a one-line hint. Simple words, honest numbers.

> Working with an AI assistant? Tell it to read `skills/README.md` first and load the one skill this assignment names on its card.

## 1. My lane (or freestyle) and why

*Name your lane — or say 'freestyle' and describe your own question. One short paragraph: why this one?*

**Ranking Signal Analysis Lane**
- I chose this lane to explore whether word count is associated with search visibility and user engagement. The available dataset provides relevant content and engagement metrics that can be used to identify meaningful patterns.

In [8]:
# This cell is for CODE (numbers, a query, a check).
# Write your text answer in the cell ABOVE this one — typing sentences here breaks Run All.


## 2. The question: decision, action, cost of a wrong call

*What decision does your work improve? Who acts on it? What does a wrong recommendation cost?*

**Question**
- Is word count associated with search visibility (impressions) and user engagement (scroll rate)?

**Decision**
- Decide whether content length should be considered when creating or updating articles to improve search visibility and user engagement.

**Who acts on it?**
- Content writers, editors, and SEO/content strategists.

**Action**
- If a meaningful relationship exists, they can adjust article length or structure (e.g., breaking up long articles, adding headings, or keeping content concise) based on the findings.

**Cost of a Wrong Recommendation**
- If the relationship is misunderstood, the team may shorten valuable articles or unnecessarily lengthen content, reducing user engagement and wasting content creation effort.

In [9]:
# This cell is for CODE (numbers, a query, a check).
# Write your text answer in the cell ABOVE this one — typing sentences here breaks Run All.


## 3. Quick look at the data (2-3 real numbers)

*Load the starter CSV below and show 2-3 real numbers that make your lane look worth the next 7 weeks.*

In [10]:
# This cell is for CODE (numbers, a query, a check).
# Write your text answer in the cell ABOVE this one — typing sentences here breaks Run All.
import pandas as pd

df = pd.read_csv("./sample_data/content_refresh_anonymized.csv")

print("Rows:", len(df))
print("Columns:", len(df.columns))

print("\nWord Count")
print(df["word_count"].describe())

print("\nImpressions")
print(df["impressions_90d"].describe())

print("\nScroll Rate")
print(df["scroll_rate"].describe())

Rows: 30000
Columns: 44

Word Count
count    22301.000000
mean      3107.760325
std       1452.382598
min          8.000000
25%       2413.000000
50%       2877.000000
75%       3666.000000
max       9546.000000
Name: word_count, dtype: float64

Impressions
count     30000.000000
mean       5200.366300
std       16838.019547
min           1.000000
25%          81.000000
50%         731.000000
75%        3615.250000
max      517715.000000
Name: impressions_90d, dtype: float64

Scroll Rate
count    29875.000000
mean        18.212921
std         29.472768
min          0.000000
25%          0.000000
50%          5.000000
75%         23.530000
max        300.000000
Name: scroll_rate, dtype: float64


The starter dataset contains 30,000 pages and 44 variables. Pages have an average word count of 3,108 words, an average of 5,200 impressions, and an average scroll rate of 18.2%, providing sufficient data to explore the relationship between content length, search visibility, and user engagement.

## 4. Careful words: what I can and can't claim

*Write what your work will be able to say (observed, directional, decision-support) — and what it never will (causal proof, 'predicting Google').*

This analysis will identify observed relationships that support content decisions. It will not prove causation or predict Google's ranking algorithm.

In [11]:
# This cell is for CODE (numbers, a query, a check).
# Write your text answer in the cell ABOVE this one — typing sentences here breaks Run All.


## Self-check

Before you submit, confirm each line honestly:

- [x] Every section above is filled — markdown thinking AND the code that backs it
- [x] The notebook runs top to bottom with no errors (Runtime → Run all)
- [x] No client names, URLs, or private queries anywhere
- [x] My claims use careful words: observed, measured, directional, decision-support
- [x] Committed to my repo under `work/notebooks/` — then submit your repo URL on the card. Done.